staging/src/k8s.io/apiserver/pkg/authentication/token/cache/cached_token_authenticator_test.go

```diff
@@ -547,3 +550,109 @@ func withAudit(ctx context.Context) context.Context {
 	ac.Event = &auditinternal.Event{Level: auditinternal.LevelMetadata}
 	return ctx
 }
+
+func TestUnsafeConversions(t *testing.T) {
+	t.Parallel()
+
+	// needs to be large to force allocations so we pick a random value between 1024 and 2048
+	// we use a multiple of sha256.Size so that the math below is simple
```

Comment:
but does it work for lengths L such that L % 32 != 0?

Reply:
[Fixed](https://github.com/kubernetes/kubernetes/compare/58ff7aab35333812213f49bbae05ded58aa50cf9..aa80f8fb856bb2b645c90457f9b1dd75e4e57c73?w=1).